### DISCUSSION TITLE
Assign role during registration process

### DISCUSSION THREAD

**Discussion by jordantsap** [filamentphp/filament#14664] [Help] [answered]
### Package

Panel builder

### Package Version

3.2

### How can we help you?

I have this code protected function getForms(): array
    {
        return [
            'form' => $this->form(
                $this->makeForm()
                    ->schema([
                        $this->getNameFormComponent(),
                        $this->getEmailFormComponent(),
                        $this->getPasswordFormComponent(),
                        $this->getPasswordConfirmationFormComponent(),
                        $this->getRoleFormComponent(),
                    ])
                    ->statePath('data'),
            ),
        ];
    }

    protected function getRoleFormComponent(): Component
    {
        return Select::make('role')
            ->options(Role::whereIn('id', [4, 5,6,7])
                ->get()
                ->pluck('name', 'id'))
            ->default('3')
            ->required();
    }

in register class of             ->registration(Register::class)
and i want to assign the role that is selected from the form, something like controller function.
How to do it

**Comment by leandrocfe** [comment]
Is role a [relationship](https://filamentphp.com/docs/3.x/forms/fields/select/#integrating-with-an-eloquent-relationship) ?

**Reply by jordantsap** [reply]
it is connected with the spatie/permissions package
Role extends \Spatie\Permission\Models\Role

**Reply by leandrocfe** [reply, marked as answer]
use

```php
Select::make('roles')
    ->multiple()
    ->relationship('roles', 'name', modifyQueryUsing: function ($query) {
        $query->whereIn('id', [4, 5, 6, 7]);
    })
    ->preload()
    ->default([4])
```
